tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -113,34 +114,59 @@ def histogram(name, data, step=None, buckets=None, description=None):
         or tf.summary.summary_scope
     )
 
+    # Try to capture current name scope so we can re-enter it below within our
+    # histogram_summary helper. We do this to avoid having the `tf.cond` below
+    # insert an extra `cond` into the tag name.
```

Comment:
This is the part I understand the least in the PR. Is it correct that without this, our summary would have tag like `some/preexsting/tag/cond/histogram_summary` when we want `some/preexsting/tag/histogram_summary` and achieve this with `tf.name_scope("") + current_scope`? If this is true, wow, `name_scope("")` is interesting.

Reply:
Yeah, that's exactly the issue. Using `tf.name_scope("")` will "reset" the active scope to the root level, which is why this works. I guess technically that's not documented in the eager API, but it is still documented for [`Graph.name_scope`](https://www.tensorflow.org/api_docs/python/tf/Graph#name_scope) and it's how `name_scope` has always handled empty strings:

https://github.com/tensorflow/tensorflow/blob/a4dfb8d1a71385bd6d122e4f27f86dcebb96712d/tensorflow/python/framework/ops.py#L6669-L6670